### ISSUE TITLE
Commit history is missing

### ISSUE THREAD

**Issue by DavidBuchanan314** [annson24/SXTools#2]
Where did it go? What sneaky changes are you hiding? :stuck_out_tongue:

**Comment by aveao** [comment]
Changes:
- Added brick code
- Added ~~stolen~~ completely original code, so no credit needs to be given
- Now paid and proprietary!

**Comment by ThatNerdyPikachu** [comment]
I mean, they give credit!!!111!!!*$*=*×

**Comment by Brawl345** [comment]
They commit history is not required to comply with the license, neither need modifications ("Licensed works, modifications, and larger works may be distributed under different terms and without source code. "). Not trying to defend this, but it is how it is, so no issue there.

**Comment by Lethaltail** [comment]
Commit history is in the fork, they went and plopped it all into a new non-fork repo for some ungodly reason.

**Comment by DavidBuchanan314** [comment]
@Brawl345 Correct, there is no *legal* issue here, but that doesn't mean it isn't an issue.

**Comment by aveao** [comment]
Well, you're not TX staff so I can remove my anti-tx comments partially here:

![](https://i.hackforjes.us/i/zwfu.png)

(still not happy about helping an official TX tool but eh)

Here have some git tips on restoring history:

---

When you have a repo locally that you want to push to a whole new repo:
- create an empty repo on github's site with no `README.md` or `LICENSE` or `.gitignore` (on Gitlab you don't need this, it creates repo on push)
- if this repo is pointing at a repo already (you cloned it from one), do `git remote set-url origin https://<your git site>/<your username>/<new project>.git`, if it's not pointing at anything, do the same but replace `set-url` with `add`
- do `git push --set-upstream origin <branch you want to push to, probably master>`, you should be done now.

In this case you already fucked up once and pushed code without history, so it'll fail to push with the steps above. However, if you do second step and just do `git push --force --set-upstream origin master` instead of third step, it should work.

---

Can you get TX to give proper credit on their site too, to NXLoader (if @DavidBuchanan314 wants it ofc)?
